core-bundle/src/Resources/contao/models/ContentModel.php

```diff
@@ -425,6 +425,9 @@ public static function findPublishedByPidAndTable($intPid, $strParentTable, arra
 			$arrColumns[] = "$t.invisible='' AND ($t.start='' OR $t.start<='$time') AND ($t.stop='' OR $t.stop>'$time')";
 		}
 
+		// Skip unsaved elements (see #2708)
+		$arrColumns[] = "NOT ($t.tstamp=0 AND $t.invisible='')";
```

Comment:
@contao/developers Why exactly have we concluded that we need to check both the timestamp and the invisible field? Shouldn't we ignore all content elements with `tstamp=0`?

Reply:
Yeah I think the additional `invisible` check is superfluous there 🤔

Reply:
Yes, the `invisible` check is already done 4 lines above so we only have to check for `tstamp!=0` IMO.